test/test_util_filenaming.py

```diff
@@ -182,3 +184,64 @@ def test_move_ensure_casing(self):
             move_ensure_casing(file_path, target_path)
             files = os.listdir(d)
             self.assertIn('FOO', files)
+
+
+class MakeSavePathTest(PicardTestCase):
+
+    def test_replace_trailing_dots(self):
+        path = 'foo./bar.'
+        self.assertEqual(path, make_save_path(path))
+        self.assertEqual('foo_/bar_', make_save_path(path, win_compat=True))
+
+    def test_replace_leading_dots(self):
+        path = '.foo/.bar'
+        self.assertEqual('_foo/_bar', make_save_path(path))
+
+    @unittest.skipUnless(IS_MACOS, "macOS test")
```

Comment:
See my comment above about `macos_compat`, this doesn't have to depend on test env.

Reply:
Good idea, done that